astroid/decorators.py

```diff
@@ -95,22 +95,22 @@ def wrapped(node, context=None, _func=func, **kwargs):
 
         yielded = set()
         generator = _func(node, context, **kwargs)
-        while True:
-            try:
-                res = next(generator)
-            except StopIteration as error:
-                if error.args:
-                    return error.args[0]
-                return None
-
-            # unproxy only true instance, not const, tuple, dict...
-            if res.__class__.__name__ == "Instance":
-                ares = res._proxied
-            else:
-                ares = res
-            if ares not in yielded:
-                yield res
-                yielded.add(ares)
+
+        try:
+            for res in generator:
+                # unproxy only true instance, not const, tuple, dict...
+                if res.__class__.__name__ == "Instance":
+                    ares = res._proxied
+                else:
+                    ares = res
+                if ares not in yielded:
+                    yield res
+                    yielded.add(ares)
+        except StopIteration as error:
```

Comment:
Can the stop iteration happen anymore while using a for loop ? Shouldn't this be a for else instead ?